include/ada/url_base.h

```diff
@@ -73,55 +73,55 @@ namespace ada {
      * Inspired after servo/url
      * @see https://github.com/servo/rust-url/blob/b65a45515c10713f6d212e6726719a020203cc98/url/src/quirks.rs#L31
      */
-    [[nodiscard]] ada_really_inline ada::url_components get_components() noexcept;
+    [[nodiscard]] virtual ada_really_inline ada::url_components get_components() noexcept = 0;
```

Comment:
It is a bit unfortunate that this function returns a *copy* of ada::url_components in the instances when the component is already computed. It makes for an unnecessary copy.

Reply:
We don't even need this at the moment. We can just remove it.